pkg/roachpb/span_config.proto

```diff
@@ -166,56 +166,99 @@ message SpanConfig {
+
+	// TargetTenantID is the ID of the tenant that the associated system span
+	// configuration applies to. If the host tenant is the source then the
+	// system span configuration applies over all ranges in the system (including
+	// those belonging to secondary tenants).
+	//
+	// Secondary tenants are only allowed to target themselves. The host tenant
+	// may use this field to target a specific secondary tenant.
+	TenantID TargetTenantID = 2 [(gogoproto.nullable) = false];
+}
+
+// SpanConfigTarget specifies the target of an associated span configuration.
+message SpanConfigTarget {
+  oneof union {
+    // Span is a keyspan that a span config is said to apply over.
+    Span span = 1;
+    // SystemSpanConfigTarget specifies the target of a system span
+    // configuration.
+    SystemSpanConfigTarget system_span_config_target = 2;
+  }
+}
+
 // SpanConfigEntry ties a span to its corresponding config.
 message SpanConfigEntry {
-  // Span is the keyspan the config is said to apply over.
-  Span span = 1 [(gogoproto.nullable) = false];
+  reserved 1;
 
-  // Config is the set of attributes that apply over the corresponding keyspan.
+  // Target specifies the target (keyspan(s)) the config applies over.
+  SpanConfigTarget target = 3 [(gogoproto.nullable) = false];
+
+  // Config is the set of attributes that apply over the corresponding target.
   SpanConfig config = 2 [(gogoproto.nullable) = false];
 };
 
-// GetSpanConfigsRequest is used to fetch the span configurations over the
-// specified keyspans.
+// GetSpanConfigsRequest is used to fetch the span configurations and system
+// span configurations.
 message GetSpanConfigsRequest {
-  // Spans to request the configurations for. The spans listed here are not
-  // allowed to overlap with one another.
-  repeated Span spans = 1 [(gogoproto.nullable) = false];
+  reserved 1;
+
+  // Targets to request configurations for. The targets listed here are not
+  // allowed to be duplicated/overlap with one another.
+  repeated SpanConfigTarget targets = 3 [(gogoproto.nullable) = false];
 };
 
-// GetSpanConfigsResponse lists out the span configurations that overlap with
-// the requested spans.
+// GetSpanConfigsResponse lists out the span configurations and system span
+// configurations that have been requested.
 message GetSpanConfigsResponse {
   // SpanConfigEntries capture the span configurations over the requested spans.
   // The results for each Span in the matching GetSpanConfigsRequest are
   // flattened out into a single slice, and follow the same ordering. It's
   // possible for there to be no configurations for a given span; there'll
   // simply be no entries for it.
-  repeated SpanConfigEntry span_config_entries = 1 [(gogoproto.nullable) = false];
+  //
+  // Any system span configurations set by the tenant are also returned if
+  // requested.
+  repeated SpanConfigEntry span_config_entries = 2 [(gogoproto.nullable) = false];
```

Comment:
just for my understanding, why do we need to bump the ID on this field? The wire type is still the same right?